python/ray/dag/compiled_dag_node.py

```diff
@@ -438,30 +455,50 @@ def _compute(self, class_handle) -> bool:
             # Propagate it and skip the actual task. We don't need to wrap the
             # exception in a RayTaskError here because it has already been wrapped
             # by the previous task.
-            self.set_intermediate_buffer(exc)
+            self.set_intermediate_buffer(operation, exc)
             return False
 
+        channel_results = []
+        for entry in input_data:
+            if isinstance(entry, GPUFuture):
+                channel_result = entry.wait()
+            else:
+                channel_result = entry
```

Comment:
I think the interface would be cleaner if the input data is always a future type, so that we always call `.wait()` here (which might be a no-op if the data is already ready).

Reply:
Done